### PR TITLE
Performance Caching Current Locale

### PR DESCRIPTION
**Pull request by thomaslow**
Contributes to:
- https://github.com/kitodo/kitodo-production/issues/3533
- https://github.com/kitodo/kitodo-production/issues/4195
- https://github.com/kitodo/kitodo-production/issues/4783

On many occasions, e.g., when translating messages, the method [LocaleHelper.getCurrentLocale](https://github.com/kitodo/kitodo-production/blob/6b5bb5e2e5fc4631e28ea33b753bfa8e68105c0d/Kitodo/src/main/java/org/kitodo/production/helper/LocaleHelper.java#L57) is called. If a user is logged in, this request will trigger a database query for the current language setting of the user, see [Line 84](https://github.com/kitodo/kitodo-production/blob/6b5bb5e2e5fc4631e28ea33b753bfa8e68105c0d/Kitodo/src/main/java/org/kitodo/production/helper/LocaleHelper.java#L84). Naturally, issuing a database query for each `#{msgs.something}` will reduce performance significantly.

In theory, this database query could be cached by Hibernate. Unfortunately, Hibernate caching is currently not working in Kitodo:
- the Hibernate first-level cache is not used, because Hibernate sessions do not persist for longer than a single database query, see most methods of [BaseDAO](https://github.com/kitodo/kitodo-production/blob/6ce1ce5c60cb6d70d5e6b2d8b9b31ea645e58bb7/Kitodo-DataManagement/src/main/java/org/kitodo/data/database/persistence/BaseDAO.java), and the first-level cache is bound to the current Hibernate session
- the Hibernate second-level cache is enabled globally, but requires the annotation `@org.hibernate.annotations.Cache` for each Bean, which are missing, see [Documentation](https://docs.jboss.org/hibernate/orm/5.4/userguide/html_single/Hibernate_User_Guide.html#caching-config): 
```
By default, entities are not part of the second level cache and we recommend you to stick to this setting.
[...]
Use the @org.hibernate.annotations.Cache annotation for this purpose.
```

Even with database caching enabled, one could argue that each `#{msgs.something}` should still not require a full re-evaluation of the Hibernate query framework. In my opinion, the current user locale should be cached in memory for each request.

The proposed pull request caches `LocaleHelper.getCurrentLocale` via a simple request scoped cache implementation using `FacesContext`, meaning the cache is cleared after each HTTP request. Caching improves performance significantly (together with #5095):

- Dashboard from 500ms to 350ms (#5095) to 250ms (both pull requests)
- List of 100 processes from 3100ms to 1400ms (#5095) to 800ms (both pull requests)
- Metadataeditor backend from ~4000ms to ~2600ms (#5095) to ~1900ms (both pull requests)

### Drawbacks

Caching via `FacesContext` could be argued to not be a very clean way of implementing this. Ideally, such information should be cached via beans that are annotated with `@RequestScope`. However, I'm not aware of any other way to introduce request scope to low-level static service methods like this. I suppose this is a JSF architectural design issue. Any suggestions are welcome.

### CODE REVIEW

File: Kitodo/src/main/java/org/kitodo/production/helper/cache/RequestScopeCacheHelper.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * <p>If this cache is used outside of a JSF request context, it will simply re-evaluate the supplier 
+ * function upon every request.</p>
+ * 
+ */
+public class RequestScopeCacheHelper {
+
+    private static final String ATTRIBUTE_NAME = "RequestScopeCacheHelper";
+
+    @SuppressWarnings("unused")
+    private static final Logger logger = LogManager.getLogger(RequestScopeCacheHelper.class);
```
Comment (markusweigelt) [inline Kitodo/src/main/java/org/kitodo/production/helper/cache/RequestScopeCacheHelper.java:47]: Do we need this logger? If not then I suggest to remove it.

**Review by markusweigelt — CHANGES_REQUESTED** [review]
Great idea to cache within the RequestScrope. From my side the code and documentation fits except of the small comment.

File: Kitodo/src/main/java/org/kitodo/production/helper/cache/RequestScopeCacheHelper.java (hunk 1 above)
Comment (thomaslow) [reply Kitodo/src/main/java/org/kitodo/production/helper/cache/RequestScopeCacheHelper.java]: It's not needed. Removed

**Review by markusweigelt — APPROVED** [review]

File: Kitodo/src/main/java/org/kitodo/production/helper/cache/RequestScopeCacheHelper.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Supplier;
+
+import javax.faces.context.FacesContext;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/production/helper/cache/RequestScopeCacheHelper.java:22]: As there is no Logger anymore this imports are not needed anymore.

Comment (markusweigelt) [reply Kitodo/src/main/java/org/kitodo/production/helper/cache/RequestScopeCacheHelper.java]: I had overlooked thx

**Pull request merged**